File: docs-examples/codecs-java/src/test/java/io/micronaut/configuration/lettuce/docs/RedisClientCommands.java
package io.micronaut.configuration.lettuce.docs;

import io.lettuce.core.api.StatefulRedisConnection;
import io.lettuce.core.api.sync.RedisCommands;
import jakarta.inject.Inject;

public final class RedisClientCommands {

    // tag::commands[]
    @Inject StatefulRedisConnection<String, String> connection;

    void commands() {
        RedisCommands<String, String> commands = connection.sync();
        commands.set("foo", "bar");
        commands.get("foo");
    }
    // end::commands[]
}

File: docs-examples/codecs-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/RedisClientCommands.kt
package io.micronaut.configuration.lettuce.docs

import io.lettuce.core.api.StatefulRedisConnection
import io.lettuce.core.api.sync.RedisCommands
import jakarta.inject.Inject

class RedisClientCommands {

    // tag::commands[]
    @Inject
    lateinit var connection: StatefulRedisConnection<String, String>

    fun commands() {
        val commands: RedisCommands<String, String> = connection.sync()
        commands.set("foo", "bar")
        commands.get("foo")
    }
    // end::commands[]
}
